packages/instrumentations/src/index.ts

```diff
@@ -18,136 +18,212 @@ import {
+import { type BunyanInstrumentation } from "@opentelemetry/instrumentation-bunyan"
+import { type CassandraDriverInstrumentation } from "@opentelemetry/instrumentation-cassandra-driver"
+import { type ConnectInstrumentation } from "@opentelemetry/instrumentation-connect"
+import { type CucumberInstrumentation } from "@opentelemetry/instrumentation-cucumber"
+import { type DataloaderInstrumentation } from "@opentelemetry/instrumentation-dataloader"
+import { type DnsInstrumentation } from "@opentelemetry/instrumentation-dns"
+import { type ExpressInstrumentation } from "@opentelemetry/instrumentation-express"
+import { type FastifyInstrumentation } from "@opentelemetry/instrumentation-fastify"
+import { type FsInstrumentation } from "@opentelemetry/instrumentation-fs"
+import { type GenericPoolInstrumentation } from "@opentelemetry/instrumentation-generic-pool"
+import { type GraphQLInstrumentation } from "@opentelemetry/instrumentation-graphql"
+import { type GrpcInstrumentation } from "@opentelemetry/instrumentation-grpc"
+import { type HapiInstrumentation } from "@opentelemetry/instrumentation-hapi"
+import { type HttpInstrumentation } from "@opentelemetry/instrumentation-http"
+import { type IORedisInstrumentation } from "@opentelemetry/instrumentation-ioredis"
+import { type KnexInstrumentation } from "@opentelemetry/instrumentation-knex"
+import { type KoaInstrumentation } from "@opentelemetry/instrumentation-koa"
+import { type LruMemoizerInstrumentation } from "@opentelemetry/instrumentation-lru-memoizer"
+import { type MemcachedInstrumentation } from "@opentelemetry/instrumentation-memcached"
+import { type MongoDBInstrumentation } from "@opentelemetry/instrumentation-mongodb"
+import { type MongooseInstrumentation } from "@opentelemetry/instrumentation-mongoose"
+import { type MySQLInstrumentation } from "@opentelemetry/instrumentation-mysql"
+import { type MySQL2Instrumentation } from "@opentelemetry/instrumentation-mysql2"
+import { type NestInstrumentation } from "@opentelemetry/instrumentation-nestjs-core"
+import { type NetInstrumentation } from "@opentelemetry/instrumentation-net"
+import { type PgInstrumentation } from "@opentelemetry/instrumentation-pg"
+import { type PinoInstrumentation } from "@opentelemetry/instrumentation-pino"
+import { type RedisInstrumentation as RedisInstrumentationV2 } from "@opentelemetry/instrumentation-redis"
+import { type RedisInstrumentation as RedisInstrumentationV4 } from "@opentelemetry/instrumentation-redis-4"
+import { type RestifyInstrumentation } from "@opentelemetry/instrumentation-restify"
+import { type RouterInstrumentation } from "@opentelemetry/instrumentation-router"
+import { type SocketIoInstrumentation } from "@opentelemetry/instrumentation-socket.io"
+import { type TediousInstrumentation } from "@opentelemetry/instrumentation-tedious"
+import { type WinstonInstrumentation } from "@opentelemetry/instrumentation-winston"
```

Comment:
I think I get your epiphany--just import the type here (cheap and fast), these are used to build the list of instrumentation modules that at time of `getInstrumentations` will be checked whether enabled in config and actually be loaded? Very cool and it seems this optimization applies to non-lambda environment too!

Dumb noob question, it seems importing type is a "newish" typescript feature, does that translate into anything impacting the Node.js version that we support?

Reply:
Type imports actually don't translate to anything in JavaScript, all of these statements are dropped when transpiling the code !